### PR TITLE
Feature/tesb 24591 packaging refactoring

### PR DESCRIPTION
**Pull request by jgreffe**

### CODE REVIEW

File: examples/tesb/library-service/README-OSGi.txt [hunk 1]
```diff
@@ -45,7 +45,7 @@ a) Without Service Registry
 
       tesb:start-aux-store
 
-      feature:repo-add mvn:org.talend.esb.examples.library-service/library-features/7.2.1-SNAPSHOT/xml
+      feature:repo-add mvn:org.talend.esb.examples.library-service/library-features/7.3.1-SNAPSHOT/xml
```
Comment (xldai) [inline examples/tesb/library-service/README-OSGi.txt]: better use <version> placeholder like what other examples did, otherwize it has to be changed every time when switching version.

File: pom.xml [hunk 2]
```diff
@@ -362,76 +377,14 @@
...
-                                <suppressionsLocation>${cxf.resources.base.path}cxf-checkstyle-suppressions.xml
-                                </suppressionsLocation>
-                                <!--failsOnError>true</failsOnError-->
-                                <!-- http://jira.codehaus.org/browse/MCHECKSTYLE-62 -->
-                                <!--aggregate>true</aggregate-->
-                            </configuration>
-                        </plugin>
-                    </reportPlugins>
-                </configuration>
-            </plugin>
```
Comment (xldai) [inline pom.xml]: what's the good reason you removed the maven-site-plugin?

File: talend-esb/src/main/descriptors/assembly.xml [hunk 3]
```diff
@@ -501,267 +497,267 @@
...
+            <outputDirectory>./container/system/org/apache/karaf/config/org.apache.karaf.config.core/${karaf.version}.tesb1/</outputDirectory>
             <fileMode>0644</fileMode>
         </file>
         <!-- patch for decanter bundles (TESB-24788) -->
         <file>
             <source>${basedir}/target/dependencies/apache-karaf-decanter-${decanter.version}-features.xml</source>
-            <outputDirectory>/container/system/org/apache/karaf/decanter/apache-karaf-decanter/${decanter.version}/</outputDirectory>
+            <outputDirectory>./container/system/org/apache/karaf/decanter/apache-karaf-decanter/${decanter.version}/</outputDirectory>
             <fileMode>0644</fileMode>
             <lineEnding>unix</lineEnding>
```
Comment (xldai) [inline talend-esb/src/main/descriptors/assembly.xml]: with only one assembly.xml for both win and linux, now have no chance to specify the lineEnding:
"&lt;lineEnding&gt;dos&lt;/lineEnding&gt;

File: pom.xml (hunk 2 above)
Comment (jgreffe) [reply pom.xml]: There was a WARNING during build, therefore it's been moved to <reporting>, not removed.

File: talend-esb/src/main/descriptors/assembly.xml (hunk 3 above)
Comment (jgreffe) [reply talend-esb/src/main/descriptors/assembly.xml]: Don't get the issue here.
Karaf provides distributions with same line endings no matter the target.

File: examples/tesb/library-service/README-OSGi.txt (hunk 1 above)
Comment (jgreffe) [reply examples/tesb/library-service/README-OSGi.txt]: Yes agree.

File: talend-esb/src/main/descriptors/assembly.xml (hunk 3 above)
Comment (xldai) [reply talend-esb/src/main/descriptors/assembly.xml]: then we may remove these lines totally?
"&lt;lineEnding&gt;unix&lt;/lineEnding&gt;

Comment (jgreffe) [reply talend-esb/src/main/descriptors/assembly.xml]: No, some files need to be processed depending on developers git configuration :/
If ok, I'd leave it as is.

Comment (xldai) [reply talend-esb/src/main/descriptors/assembly.xml]: Okay (even not fully convinced)

**Pull request merged**